src/controllers/organizationController.ts

```diff
@@ -196,61 +209,134 @@ export default class OrganizationController {
-          );
-          if (errorsArray.length > 0) {
-            return ResponseErrorInternal(
-              "An error occurred while signing documents"
+          if (
+            organizationInstance.registrationStatus ===
+            OrganizationRegistrationStatusEnum.REGISTERED
+          ) {
+            return ResponseErrorConflict(
+              "The required documents have already been sent and countersigned"
             );
           }
-          try {
-            await this.emailService.send({
-              attachments: [
-                {
-                  filename: "contratto.pdf",
-                  path: signedContractPath
-                },
-                {
-                  filename: `delega-${user.fiscalCode.toLocaleLowerCase()}.pdf`,
-                  path: signedMandatePath
-                }
-              ],
-              html:
-                localeIt.organizationController.sendDocuments.registrationEmail
-                  .content,
-              subject:
-                localeIt.organizationController.sendDocuments.registrationEmail
-                  .subject,
-              text:
-                localeIt.organizationController.sendDocuments.registrationEmail
-                  .content,
-              to: organizationInstance.pec
-            });
-            return ResponseNoContent();
-          } catch (error) {
-            log.error("An error occurred sending email. %s", error);
-            return ResponseErrorInternal("An error occurred sending email");
-          }
+          return this.signAndSendDocuments(
+            user.fiscalCode,
+            organizationInstance
+          );
         }
       );
     });
   }
 
+  private async deleteAssociatedPreDraftOrganization(
+    userEmail: string
+  ): Promise<Option<IResponseErrorInternal | IResponseErrorConflict>> {
+    const errorOrMaybeOrganizationInstance = await getOrganizationInstanceFromDelegateEmail(
+      userEmail
+    );
+    if (errorOrMaybeOrganizationInstance.isLeft()) {
+      log.error(
+        "An error occurred reading data from db. %s",
+        errorOrMaybeOrganizationInstance.value
+      );
+      return some(
+        ResponseErrorInternal("An error occurred reading data from DB")
+      );
+    }
+    const organizationInstance = errorOrMaybeOrganizationInstance.value.toNullable();
+    if (organizationInstance) {
+      if (
+        organizationInstance.registrationStatus !==
+        OrganizationRegistrationStatusEnum.PRE_DRAFT
+      ) {
+        return some(
+          ResponseErrorConflict(
+            "There is already a registered organization associated to your account"
+          )
+        );
+      } else {
+        // The organization already associated to the user
+        // is still in draft or pre-draft status,
+        // so its registration process must be canceled
+        const maybeError = await deleteOrganization(organizationInstance);
+        if (maybeError.isSome()) {
```

Comment:
not needed now but usually prefer map / fold over if conditions (this lead to a shorter / safer code): 

```ts
return maybeError.map(error => { 
  log.error(...); 
  return ResponseErrorInternal(..) 
});
```